yt_dlp/extractor/pornbox.py

```diff
@@ -0,0 +1,137 @@
+from .common import InfoExtractor
+from ..utils import (
+    parse_duration,
+    traverse_obj,
+    str_or_none,
+    int_or_none,
+    strip_or_none,
+    parse_iso8601,
+    qualities
+)
+
+
+class PornboxIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?pornbox\.com/application/watch-page/(?P<id>[0-9]+)'
+    _TESTS = [{
+        'url': 'https://pornbox.com/application/watch-page/426095',
+        'md5': '0cc35f4d4300bc8e0496aef2a9ec20b9',
+        'info_dict': {
+            'id': '426095',
+            'ext': 'mp4',
+            'title': 'md5:96b05922d74be2be939cf247294f26bd',
+            'uploader': 'Giorgio Grandi',
+            'timestamp': 1686088800,
+            'upload_date': '20230606',
+            'age_limit': 18,
+            'availability': 'needs_auth',
+            'duration': 1767,
+            'cast': ['Tina Kay', 'Neeo', 'Mr. Anderson', 'Thomas Lee', 'Brian Ragnastone', 'Rycky Optimal'],
+            'tags': 'count:37',
+            'thumbnail': r're:^https?://cdn-image\.gtflixtv\.com.*\.jpg.*$'
+        }
+    }, {
+        'url': 'https://pornbox.com/application/watch-page/216045',
+        'md5': '56cb78bb1d8e0d2dad8b75c278c20098',
+        'info_dict': {
+            'id': '216045',
+            'ext': 'mp4',
+            'title': 'md5:3e48528e73a9a2b12f7a2772ed0b26a2',
+            'description': 'md5:3e631dcaac029f15ed434e402d1b06c7',
+            'uploader': 'VK Studio',
+            'timestamp': 1618264800,
+            'upload_date': '20210412',
+            'age_limit': 18,
+            'availability': 'premium_only',
+            'duration': 2710,
+            'cast': ['Bella Grey', 'Oliver Trunk'],
+            'tags': 'count:29',
+            'thumbnail': r're:^https?://cdn-image\.gtflixtv\.com.*\.jpg.*$',
+            'subtitles': 'count:6'
+        },
+        'params': {
+            'skip_download': True,
+            'ignore_no_formats_error': True
+        },
+        'skip': 'Only for subscribers'
```

Comment:
You don't need to skip if doing ignore_no_formats_error, no?

Reply:
I tried, but the test aborts with the login (raise_login_required) error

Reply:
I think I have found the problem. `raise_login_required()` calls `self.report_warning()`, which is modified in line 51 of test_download.py to throw an `ExtractorError`. This is then caught in line 160. Based on the comment in line 50 ("Don't accept warnings during tests"), I would assume that this is intended behaviour and nothing I can work around here.

I have temporarily commented out the part where the `ExtractorError` is thrown and then the test passes.

Reply:
I think it needs this
```suggestion
        'expected_warnings': ['This video is only available for registered users'],
        'skip': 'Only for subscribers'
```